Wemos_Modules/Door/Door.ino

```diff
@@ -0,0 +1,215 @@
+/*
+ * HHS PJSD 2021
+ *
+ * Groep 1
+ *
+ * Marko Estrada Rodriguez
+ * Wouter de Bruijn
+ * Casper Bastiaan-Net
+ * Maurice Tacx
+ * Steven Cowan
+ *
+ * Wemos I/O Server
+ *
+ * Wemos module connecting to a wifi network and creating a TCP server.
+ * Can receive TCP server commands to set or get data.
+ * Can use IO from WIB Board
+ *
+ * NOTE: using library ArduinoJson 6.18.5
+ */
+
+#include <Wire.h>
+#include <Servo.h>
+#include <ArduinoJson.h>
+#include <ESP8266WiFi.h>
+#include <ArduinoWiFiServer.h>
+
+#define WIFI_SSID "G41C control node"
+#define WIFI_PASSWD "g41cstandsforgalc"
+#define PORT 8080
+#define ANALOG_IC_ADDR 0x36
+#define DIGITAL_IC_ADDR 0x38
+#define DIGITAL_IC_IN 0x00
+#define DIGITAL_IC_OUT 0x01
+
+/*const IPAddress local_IP(172, 16, 99, 100);
+const IPAddress gateway(172, 16, 99, 1);
+const IPAddress subnet(255, 255, 255, 0);*/
+
+void configureDigitalIC();
+void configureAnalogIC();
+
+unsigned int readDigitalInputs();
+void setDigitalOutput(const int pin, const bool state);
+
+void readAnalogInput();
+void handleConnections();
+void connectWifi();
+
+Servo doorServo;
+
+bool buttonOutside = false;
+bool buttonInside = false;
+bool ledInside = false;
+bool ledOutside = false;
+int door = 0;
+byte digitalBuffer = 0;
+
+ArduinoWiFiServer server(PORT);
+
+void setup()
+{
+    // Start up Serial debug
+    Serial.begin(9600);
+
+    connectWifi();
+
+    Wire.begin();
+
+    delay(1000);
+
+    // Use pin d5 as servo output
+    doorServo.attach(D5);
+    pinMode(D5, OUTPUT);
+
+    configureDigitalIC();
+
+    // Start TCP Server
+    server.begin();
+    Serial.println("TCP Server started listening...");
+
+    Serial.end();
+}
+
+void loop()
+{
+    // Handle clients sending request to the TCP server.
+    handleConnections();
+
+    int inputData = readDigitalInputs();
+
+    buttonOutside = inputData & (1 << 0);
```

Comment:
Bitshift isn't needed, can be kept because it looks nice 💎

Reply:
im keeping it for clarity's sake.